File: qz/u2/santi/Exam U2/71-100/ExamSpeakers/src/main/java/ec/edu/espe/examspeakers/controller/SpeakerDAO.java

package ec.edu.espe.examspeakers.controller;

import ec.edu.espe.examspeakers.model.Speaker;
import java.util.List;

/**
 *
 * @author Sjean
 */
public interface SpeakerDAO {
    void create(Speaker speaker);
    Speaker read(String id);
    void update(Speaker speaker);
    void delete(String id);
    List<Speaker> listAll();
}

File: qz/u3/sigcha/qz1&2/Quizz_Speakers/Quizz_Speakers/controller/speaker_dao.py
from abc import ABC, abstractmethod
from model.speaker import Speaker
from typing import List

class SpeakerDAO(ABC):
    @abstractmethod
    def create(self, speaker: Speaker):
        pass

    @abstractmethod
    def read(self, speaker_id: str) -> Speaker:
        pass

    @abstractmethod
    def update(self, speaker: Speaker):
        pass

    @abstractmethod
    def delete(self, speaker_id: str):
        pass

    @abstractmethod
    def list_all(self) -> List[Speaker]:
        pass
